pages/hold/request/[id].tsx

```diff
@@ -0,0 +1,340 @@
+import Head from "next/head"
+import { useState, useContext, useRef, useEffect } from "react"
+import { useRouter } from "next/router"
+import {
+  Heading,
+  List,
+  Form,
+  FormField,
+  RadioGroup,
+  Radio,
+  ButtonGroup,
+  Button,
+  Banner,
+  Box,
+  SkeletonLoader,
+} from "@nypl/design-system-react-components"
+
+import Layout from "../../../src/components/Layout/Layout"
+import { FeedbackContext } from "../../../src/context/FeedbackContext"
+import {
+  PlainTextElement,
+  LinkedDetailElement,
+} from "../../../src/components/BibPage/BibDetail"
+import RCLink from "../../../src/components/Links/RCLink/RCLink"
+import {
+  SITE_NAME,
+  BASE_URL,
+  PATHS,
+  BUTTON_LINK_STYLES,
+} from "../../../src/config/constants"
+import useLoading from "../../../src/hooks/useLoading"
+
+import { fetchBib } from "../../../src/server/api/bib"
+import initializePatronTokenAuth, {
+  doRedirectBasedOnNyplAccountRedirects,
+  getLoginRedirect,
+} from "../../../src/server/auth"
+
+import Bib from "../../../src/models/Bib"
+import Item from "../../../src/models/Item"
+
+import bibDetailStyles from "../../../styles/components/BibDetails.module.scss"
+
+import type { DiscoveryBibResult } from "../../../src/types/bibTypes"
+import type { DiscoveryItemResult } from "../../../src/types/itemTypes"
+import type { ItemMetadata } from "../../../src/types/itemTypes"
+
+interface HoldRequestPropsType {
+  discoveryBibResult: DiscoveryBibResult
+  discoveryItemResult: DiscoveryItemResult
+  isAuthenticated?: boolean
+}
+
+/**
+ * The Hold Request page renders a form to place a hold on an item.
+ * It renders some basic info about the bib and item based on the bib and item ids in the url,
+ * including pickup location options rendered as radio buttons in the form.
+ */
+export default function HoldRequestPage({
+  discoveryBibResult,
+  discoveryItemResult,
+  isAuthenticated,
+}: HoldRequestPropsType) {
+  const metadataTitle = `Item Request | ${SITE_NAME}`
+
+  const bib = new Bib(discoveryBibResult)
+  const item = new Item(discoveryItemResult, bib)
+  const holdId = `${item.bibId}-${item.id}`
+
+  const [alert, setAlert] = useState(false)
+  const [formPosting, setFormPosting] = useState(false)
+  const notificationRef = useRef<HTMLDivElement>()
+  const { onOpen, setItemMetadata } = useContext(FeedbackContext)
+
+  const router = useRouter()
+  const isLoading = useLoading()
+
+  useEffect(() => {
+    if (alert && notificationRef.current) {
+      notificationRef.current.focus()
+    }
+  }, [alert])
+
+  const handleSubmit = async (e) => {
+    e.preventDefault()
+
+    try {
+      setFormPosting(true)
+      const response = await fetch(`${BASE_URL}/api/hold/request/${holdId}`, {
+        method: "POST",
+        // TODO: serialize form data
+        body: "",
+      })
+      const responseJson = await response.json()
+
+      if (response.status !== 200) {
+        console.error("Error in hold  request api response", responseJson.error)
```

Comment:
Prefix the string with `"HoldRequestPage: Error in..."` so it's easier to debug. Nitpick: extra space after "hold" and API instead of api.